lib/gonzales.ts

```diff
@@ -196,19 +196,21 @@ read_local_memory (task_t remote_task,
 }
 `;
 
-const {getApi} = require('./api');
+import {getApi} from "./api";
 
 const {pointerSize} = Process;
 
-let cachedModule = null;
+interface Gonzales {
+    [name: string]: AnyFunction;
```

Comment:
This can be narrowed down to just the API actually exposed.

Reply:
Do you mean dropping the declaration of `interface Gonzales` and replacing all the `Gonzales` by `{ [name: string]: AnyFunction; }`, such like below:

```
let cachedModule: { [name: string]: AnyFunction } | null = null;

export function getGonzales(): { [name: string]: AnyFunction } {
    if (cachedModule === null)
        cachedModule = compileModule();
    return cachedModule;
}

```

Reply:
No. I mean making the definition contain the exact properties it contains, instead of a `[name: string]: AnyFunction`, which is not very typesafe.